### PR TITLE
device agnostic pipelines testing

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?

Part of https://github.com/huggingface/transformers/issues/25654#issuecomment-1783704306

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

cc @ydshieh

### CODE REVIEW

**Comment by statelesshz** [comment]
verified with Mac-Studio

image

File: src/transformers/testing_utils.py [hunk 1]
```diff
@@ -774,7 +774,9 @@ def require_torch_gpu(test_case):
 
 def require_torch_accelerator(test_case):
     """Decorator marking a test that requires an accessible accelerator and PyTorch."""
-    return unittest.skipUnless(torch_device != "cpu", "test requires accelerator")(test_case)
+    return unittest.skipUnless(torch_device is not None and torch_device != "cpu", "test requires accelerator")(
```
Comment (statelesshz) [inline src/transformers/testing_utils.py:777]: We need this to avoid incorrect execution of PyTorch related testing suites in TensorFlow environment

**Comment by amyeroberts** [comment]
cc @ydshieh

**Comment by statelesshz** [comment]
It seems the failing test case is not related to these commits.

**Comment by statelesshz** [comment]
@ydshieh Marking this ready for review.

**Review by amyeroberts — APPROVED** [review]
Thanks for working on this and improving our tests! ❤️ 

Happy to merge once @ydshieh approves

File: tests/pipelines/test_pipelines_text_classification.py [hunk 2]
```diff
@@ -96,13 +96,11 @@ def test_small_model_pt(self):
...
     @require_torch
     def test_accepts_torch_device(self):
-        import torch
-
         text_classifier = pipeline(
             task="text-classification",
             model="hf-internal-testing/tiny-random-distilbert",
             framework="pt",
-            device=torch.device("cpu"),
+            device=torch_device,
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_text_classification.py:103]: not sure why it was hardcoded to `cpu` previously

**Review by ydshieh — APPROVED** [review]
Great! I would like to run them before merge. Will update here.

**Comment by ydshieh** [comment]
test run results look good, so merge! Thanks again @statelesshz

**Pull request merged**